components/file_system/src/thread_io.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+use crate::IOBytes;
+use crate::IOType;
+use std::cell::Cell;
+use std::fs::File;
+use std::io::BufRead;
+use std::io::BufReader;
+use std::path::PathBuf;
+use std::str::FromStr;
+
+thread_local! {
+    static IO_TYPE: Cell<IOType> = Cell::new(IOType::Other)
+}
+
+pub fn set_io_type(new_io_type: IOType) {
+    IO_TYPE.with(|io_type| {
+        io_type.set(new_io_type);
+    });
+}
+
+pub fn get_io_type() -> IOType {
+    IO_TYPE.with(|io_type| io_type.get())
+}
+
+pub(crate) fn flush_io_latency_metrics() {}
+
+pub(crate) fn fetch_io_bytes(_io_type: IOType) -> IOBytes {
```

Comment:
where it will be called

Reply:
I expect this function does what the same name function in `iosnoop` do.
But I have no idea how it should handle the param

Reply:
let's move on, now integrate it into the io rate limiter.